### PR TITLE
Max/add passable envs and bucket by key

### PR DESCRIPTION
**Pull request by maxwellsmart84**
This adds the ability to drop in your AWS creds to the RhinoCloud class which then passes them on to CF and S3.  I also add an additional parameter to getBuckets so you can filter your Object returns.  In addition I added a default empty object to functions I found using destruction.  This prevents type errors when the function is called without params.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 const CloudFormationWrapper = require('./library/cloudformation');
 const S3Wrapper = require('./library/s3');
 
-function Rhinocloud() {
-  this.cloudformation = new CloudFormationWrapper();
-  this.s3 = new S3Wrapper();
+function Rhinocloud({ accessKeyId = undefined, secretAccessKey = undefined, region = undefined } = {}) {
```
Comment (rogersgt) [inline index.js:4]: Shouldn't these default to `process.env.AWS_ACCESS_KEY_ID`, `process.env.AWS_SECRET_ACCESS_KEY`, and `process.env.AWS_REGION`?

File: library/cloudformation/index.js [hunk 2]
```diff
@@ -51,7 +56,7 @@ function CloudFormationWrapper() {
     return resp;
   }
 
-  async function deleteStack({ stackName, options={} }) {
+  async function deleteStack({ stackName, options={} } = {}) {
```
Comment (rogersgt) [inline library/cloudformation/index.js:63]: If we are going to allow an empty object, there should be a check to make sure that `stackName` is defined.

File: library/cloudformation/index.js [hunk 3]
```diff
@@ -3,11 +3,16 @@ const { CloudFormation } = require('aws-sdk');
...
+  const cf = new CloudFormation({
+    apiVersion: apiVersions.CloudFormation,
+    accessKeyId,
+    secretAccessKey,
+    region
+  });
 
   // -------------------------- API functions ---------------------------- //
-  async function changeTerminationProtection({ stackName, enableTerminationProtection }) {
+  async function changeTerminationProtection({ stackName, enableTerminationProtection } = {}) {
```
Comment (rogersgt) [inline library/cloudformation/index.js:15]: Need to check that `stackName` is defined if the function allows an empty object.

**Review by rogersgt — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (maxwellsmart84) [reply index.js:4]: you actually don't have to do that - the sdk looks for the envs automatically.

https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html

File: library/cloudformation/index.js [hunk 4]
```diff
@@ -3,11 +3,18 @@ const { CloudFormation } = require('aws-sdk');
...
+    if (!templatePath || !enableTerminationProtection) {
```
Comment (rogersgt) [inline library/cloudformation/index.js]: Should this be `stackName` instead of `templatePath`?

File: index.js (hunk 1 above)
Comment (rogersgt) [reply index.js:4]: 👍

**Review by rogersgt — CHANGES_REQUESTED** [review]

**Review by rogersgt — APPROVED** [review]

**Pull request merged**